schema/resolversTests/validation.test.js

```diff
@@ -296,58 +309,139 @@ describe("resolvers", () => {
       );
     });
 
-    it("should be able to update earliest date properties", async () => {
-      const answer = await createNewAnswer(firstPage, "Date");
-      const validation = await queryAnswerValidations(answer.id);
-      const result = await mutateValidationParameters({
-        id: validation.earliestDate.id,
-        earliestDateInput: {
-          custom: "2017-01-01",
-          offset: {
-            value: 8,
-            unit: "Months"
-          },
-          relativePosition: "After"
-        }
+    describe("Earliest", () => {
+      it("should be able to update properties", async () => {
+        const answer = await createNewAnswer(firstPage, "Date");
+        const validation = await queryAnswerValidations(answer.id);
+        const result = await mutateValidationParameters({
+          id: validation.earliestDate.id,
+          earliestDateInput: {
+            ...params,
+            custom: "2017-01-01"
+          }
+        });
+        const expected = {
+          id: validation.earliestDate.id,
+          ...params,
+          customDate: "2017-01-01",
+          previousAnswer: null
+        };
+        expect(result).toEqual(expected);
+      });
+
+      it("can update previous answer", async () => {
+        const previousAnswer = await createNewAnswer(firstPage, "Date");
+        const answer = await createNewAnswer(firstPage, "Date");
+        const validation = await queryAnswerValidations(answer.id);
+
+        const result = await mutateValidationParameters({
+          id: validation.earliestDate.id,
+          earliestDateInput: {
+            ...params,
+            entityType: "PreviousAnswer",
+            previousAnswer: previousAnswer.id
+          }
+        });
+
+        expect(result).toMatchObject({
+          entityType: "PreviousAnswer",
+          previousAnswer: {
+            id: previousAnswer.id
+          }
+        });
+      });
+
+      it("can update entity type", async () => {
+        const answer = await createNewAnswer(firstPage, "Date");
+        const validation = await queryAnswerValidations(answer.id);
+
+        const entityTypes = ["Custom", "PreviousAnswer"];
+
+        const promises = entityTypes.map(async entityType => {
+          const result = await mutateValidationParameters({
+            id: validation.earliestDate.id,
+            earliestDateInput: {
+              ...params,
+              entityType: entityType
```

Comment:
No need for the second entityType here.

```js
earliestDateInput: {
  ...params,
  entityType
}
```

Reply:
Do you think there's an eslint rule to catch this? What's this even called? I much prefer it.